src/catnap/linux/transport.rs

```diff
@@ -574,8 +577,23 @@ impl NetworkTransport for SharedCatnapTransport {
         &mut self,
         sd: &mut Self::SocketDescriptor,
         size: usize,
-    ) -> Result<(Option<SocketAddr>, DemiBuffer), Fail> {
-        self.data_from_sd(sd).pop(size).await
+    ) -> Result<(Option<SocketAddr>, ArrayVec<DemiBuffer, DEMI_SGARRAY_MAXLEN>), Fail> {
+        let total_size: usize = 0;
```

Comment:
What's the role of `total_size` in the while loop? It's value never changes.

Reply:
Ooops, I should be updating it.